leptos_dom/src/html.rs

```diff
@@ -679,6 +696,90 @@ impl<El: ElementDescriptor + 'static> HtmlElement<El> {
+        C: Into<Cow<'static, str>>,
+    {
+        #[cfg(all(target_arch = "wasm32", feature = "web"))]
+        {
+            use smallvec::SmallVec;
+
+            let el = self.element.as_ref().clone();
+
+            leptos_reactive::create_effect(
+                self.cx,
+                move |prev_classes: Option<
+                    SmallVec<[Cow<'static, str>; 4]>,
+                >| {
+                    let classes = classes_signal()
+                        .into_iter()
+                        .map(Into::into)
+                        .collect::<SmallVec<[Cow<'static, str>; 4]>>(
+                    );
+
+                    let mut new_classes = classes
+                        .iter()
+                        .flat_map(|classes| classes.split_whitespace());
+
+                    if let Some(prev_classes) = prev_classes {
+                        let mut old_classes = prev_classes
+                            .iter()
+                            .flat_map(|classes| classes.split_whitespace());
+
+                        // Remove old classes
+                        for prev_class in old_classes.clone() {
+                            if !new_classes.any(|c| c == prev_class) {
+                                el.class_list()
+                                    .remove_1(prev_class)
+                                    .unwrap_or_else(|err| {
+                                        panic!(
+                                            "failed to add class \
+                                             `{prev_class}`, error: {err:#?}"
+                                        )
+                                    });
+                            }
+                        }
+
+                        // Add new classes
+                        for class in new_classes {
+                            if !old_classes.any(|c| c == class) {
+                                el.class_list().add_1(class).unwrap_or_else(
+                                    |err| {
+                                        panic!(
+                                            "failed to remove class \
+                                             `{class}`, error: {err:#?}"
+                                        )
+                                    },
+                                );
+                            }
+                        }
+                    } else {
+                        for class in new_classes {
+                            el.class_list().add_1(class).unwrap_or_else(
+                                |err| {
+                                    panic!(
+                                        "failed to add class `{class}`, \
+                                         error: {err:#?}"
+                                    )
+                                },
+                            );
+                        }
+                    }
+
+                    classes
+                },
+            );
+        }
+
```

Comment:
I notice this method just does nothing in SSR mode. I think it should, instead, read the initial value of the signal and add it to the classes in the element. Otherwise, if there are classes here, they'll pop in once WASM is loaded and there be unstyled content beforehand.